h5netcdf/core.py

```diff
@@ -415,13 +415,17 @@ def __repr__(self):
 class File(Group):
 
     def __init__(self, path, mode='a', **kwargs):
-        self._h5file = h5py.File(path, mode, **kwargs)
+        self._closed = True
+        try:
```

Comment:
I think it would be clearer to write:
```python
try:
    self._h5file = ...
except Exception:
    self._closed = True
    raise
else:
    self._closed = False
```
Note that it's important to catch `Exception` rather than use the naked `except`, to avoid catching `KeyboardInterrupt` and the like.